mailserver/limiter_test.go

```diff
@@ -0,0 +1,95 @@
+package mailserver
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+func TestLimiterSuite(t *testing.T) {
+	suite.Run(t, new(LimiterSuite))
+}
+
+type LimiterSuite struct {
+	suite.Suite
+}
+
+func (s *LimiterSuite) TestIsAllowed() {
+	peerID := "peerID"
+	testCases := []struct {
```

Comment:
In table driven tests I like a little `info string` in the struct with the info logged in the beginning of each loop. Makes localizing of a failure easier.